### PR TITLE
Trim down Dockerfile Size

### PR DESCRIPTION
**Pull request by learnitall**
This PR trims down the Dockerfile size from ~2GB to 554 MB by finding ways to remove large dependencies. The following dependencies are removed from the Dockerfile:

* `git`, by using the GitHub API to download other dependencies instead.
* `jq` and `gcloud`, by using a python script to generate a kubeconfig for provisioned GKE clusters.
* `ansible` python package, replaced `ansible-core`, which is a stripped down version. Collections are installed as-needed.

Additionally, I've modified the Dockerfile to use the project root on the local filesystem rather than cloning the scaffolding repository, in order to help facilitate testing local changes.

### CODE REVIEW

File: roles/platform/tasks/gke.yml [hunk 1]
```diff
@@ -72,9 +74,14 @@
         - https://www.googleapis.com/auth/cloud-platform
       state: present
     register: cluster
+  
+  - name: Save cluster info to archive
```
Comment (jtaleric) [inline roles/platform/tasks/gke.yml:78]: This failed for me.

```
fatal: [localhost]: FAILED! => changed=false 
  checksum: 58f368ebfb1f4d8df6fb72a424c665bf6fce1852
  diff: []
  invocation:
    module_args:
      _original_basename: tmpksvjqimv
      attributes: null
      backup: false
      checksum: 58f368ebfb1f4d8df6fb72a424c665bf6fce1852
      content: null
      dest: /tmp/auto_perf/1652828827/cluster.json
      directory_mode: null
      follow: false
      force: true
      group: null
      local_follow: null
      mode: null
      owner: null
      remote_src: null
      selevel: null
      serole: null
      setype: null
      seuser: null
      src: /home/jtaleric/.ansible/tmp/ansible-tmp-1652829160.1815464-1898353-79276582407777/source
      unsafe_writes: false
      validate: null
  msg: Destination directory /tmp/auto_perf/1652828827 does not exist
```

Also, we should consider dropping the metadata earlier in the run to allow the `destroy=true` to execute correctly.. so lines 112-118

**Review by jtaleric — COMMENTED** [review]
Running the platform play locally failed for me. 

The container image size savings is amazing! Great stuff.

File: roles/platform/tasks/gke.yml (hunk 1 above)
Comment (learnitall) [reply roles/platform/tasks/gke.yml:87]: Ack! I moved up the creation of the archive_dir and added a `mkdir -p` in there to ensure the directory is created beforehand.

**Review by jtaleric — APPROVED** [review]
LGTM! Thanks!

**Pull request merged**